### ISSUE TITLE
Goal should not be preselected when coming from standard new site flow

### ISSUE THREAD

**Issue by autumnfjeld** [Automattic/wp-calypso#96481]
### Quick summary

See slack discussion https://a8c.slack.com/archives/C07H21B2W59/p1731690474662039?thread_ts=1731643905.223809&cid=C07H21B2W59

### Steps to reproduce

* Click new site in logged-in Sites Overview
* Go through domain and plans selection
  * I clicked Use my own domain, but did not finish this or use an actual own domain
  *  Then went back to the domain page 
* Land on Goals screen, see that a Goals is pre-selected

### What you expected to happen

* Click new site in Sites Overview or in any "generic" new site flow
* Go through domain and plans selection
* Land on Goals screen, where Goals are NOT pre-selected

### What actually happened

A Goal was preselected, I don't think it should be.

**Comment by p-jackson** [comment]
I can only reproduce this bug if I follow some steps _before_ clicking "new site" and entering the onboarding flow. I assume this is the same for you @autumnfjeld?

So my steps to reproduce are:
- Click new site in logged-in Sites Overview
- Proceed to goals screen
- Make a goals selection and click **Next**
- Using the address bar go direction to `/sites`
- Click new site in logged-in Sites Overview
- Land on goals screen, see that goals have been pre-selected

At the root of this problem is a decision: when should we clear the onboarding flow state? Interestingly different onboarding flows decide differently.
- Clear the onboarding state when the user completes the flow. [This is how the `site-setup` flow currently works](https://github.com/Automattic/wp-calypso/blob/49b0ffd708a00c4ce6cc84822995804fd5abeb1a/client/landing/stepper/declarative-flow/site-setup-flow.ts#L257)
- Clear the onboarding state when the user starts the flow. [This is how the `import-hosted-site` flow works for example](https://github.com/Automattic/wp-calypso/blob/49b0ffd708a00c4ce6cc84822995804fd5abeb1a/client/landing/stepper/declarative-flow/import-hosted-site.ts#L44)

This bug report seems to imply the user would expect the goals to be reset each time the user _enters_ the flow, but I feel I've worked on onboarding long enough to know that I can't make a big change like this without risking breaking unexpected things. For example there might be state that needs to be saved between when the user leaves checkout and enters the `site-setup` flow?

**Comment by p-jackson** [comment]
Another option was thinking of this simply as a requirement that the goals screen should always prevent the user with a blank slate, and always forget the user's previous selection. Then we wouldn't have to worry about store resets. We simply have the goals UI ignore the state of the onboarding store when it initially renders the checkboxes.

The problem with this, of always presenting the user with a blank slate, is that it'd also apply when the user clicks **Back** in the stepper. The user would choose goals, go to the next step, change their mind, go back, and they'd have to re-check all the goals again.

**Comment by inaikem** [comment]
Hey @autumnfjeld, @p-jackson 👋 I have a quick process question for you both: Can we assume issues like reported by developers are already triaged? And if so, remove the `Needs Triage` label and update the `Status` field on The One Board?

**Comment by p-jackson** [comment]
> Can we assume issues like reported by developers are already triaged? And if so, remove the Needs Triage label and update the Status field on The One Board?

Yeah probably, but maybe not always? Developer reported bugs often come up either during development or CfT, which means they're being triaged by the dev team. But sometimes I'm just reporting a bug and wouldn't expect it to privileged compared to other bug reports.

Also, I don't know exactly what happened in this case, but sometimes when I enter a bug the GH bots do a bunch of automation, assigning it to boards and giving it statuses. And I just let it do it's thing, because I don't want to second guess some automations someone else has already set up.

But in this case I removed "Needs Triage".

**Comment by inaikem** [comment]
Apologies for the late reply, @p-jackson. I pinged you in the Slack discussion here: p1732078853093839-slack-C07GZ2UA3TN

**Comment by autumnfjeld** [comment]
Sorry about the delayed response @p-jackson 

*  Only last week did I start seeing Goals pre-selected
* I've created new sites many times prior to last week via "Add new site" and never had Goals pre-selected
* I cannot say that the[ steps you outline](https://github.com/Automattic/wp-calypso/issues/96481#issuecomment-2481685741) are the same steps I went through last week. I didn't specifically do anything to reproduce, I just went to "Add new site" and then saw pre-selected Goals.
* To see if I could "clear out" the pre-selected goal:
*   I clicked "Add new site"
*  I unselected the gaol, and proceeded with no goal
* I completed onboarding
* I navigated back to /sites
*   I clicked "Add new site"
* I saw the Goals screen as expected: no goal selected

I was not able to reproduce this problem on my non-a8c account.  In my non-a8c account the Goals screen appeared as expected, nothing pre selected.

Thus, I would conclude it was a coincidence that bot @richtabor  and I ran into this same issue on the same day and that the issue is likely a fluke of our frequent site creation and unlikely to be something users are experiencing.

---------
I'm not sure any further work is needed, but to answer this philosophical question: 

> when should we clear the onboarding flow state?

I'd say after the user completes onboarding.  


-------
Thank you for investigating!

**Comment by p-jackson** [comment]
I left this issue open for a bit to see if it came up again.

> the issue is likely a fluke of our frequent site creation

I think so.
